packages/vite-node/src/server.ts

```diff
@@ -24,7 +24,27 @@ export class ViteNodeServer {
   constructor(
     public server: ViteDevServer,
     public options: ViteNodeServerOptions = {},
-  ) {}
+  ) {
+    // @ts-expect-error ssr is not typed
+    const ssrOptions = server.config.ssr
+    if (ssrOptions) {
+      options.deps ??= {}
+
+      // we don't externalize ssr, because it has different semantics in Vite
+      // if (ssrOptions.external) {
+      //   options.deps.external ??= []
+      //   options.deps.external.push(...ssrOptions.external)
+      // }
+
+      if (ssrOptions.noExternal === true) {
+        options.deps.inline = true
```

Comment:
```suggestion
        options.deps.inline ??= true
```

Inline config should have higher priority

Reply:
I am not sure that it should. What are your arguments? I think, if user decided he wants to inline everything for `ssr`, we should do the same no matter what's inside `deps.inline`, because it was a decision made by the user for his actual application.

Reply:
If we do `??=` it means `noExternal: true` will never call this `if`, because we always add our `inline` in Vitest, for example.

Reply:
`test.deps.inline` is more specific to test and to me it should always have the highest priority in Vitest if it set. Since Vitest might shares the option with the app, `ssr` might be used for the app but not the tests.

Reply:
Again, this is `vite-node`. We always append something to `deps.inline` in Vitest, so if we remove this condition here, it won't ever run. There is no reason to enable inlining in the app, and make it partial in Vitest (which is `imposible` right now, because `inline` is **always** defined by us). If you don't want something to be inline, use `deps.external`.

And again, you are saying that if user defines `test.deps.inline`, it should have higher priority, but it is impossible here, because when code goes here, we already defined/appended `inlined` in Vitest.

Reply:
> which is `imposible` right now, because `inline` is **always** defined by us

To resolve this, we might change this to be more conditional:

https://github.com/vitest-dev/vitest/blob/c79f9f91c2c1871dcf1e04013d9307dd6f30cdba/packages/vitest/src/node/config.ts#L96

- If defined `true`, stay with true
- If defined `noExternal`, but not defined `inline`, `inline` becomes `true`
- If defined `noExternal`, and defined `inline`, `inline` stays the same, and extended by us
- If not defined, defined with our inlines

Reply:
I see now. Given this is rather a niche case, I am fine either way. I'll leave the decision to you.

Reply:
> I see now. Given this is rather a niche case, I am fine either way. I'll leave the decision to you.

I'll change it to `??=`, but will alter how Vitest appends it's inlines